tests/client/test_headers.py

```diff
@@ -181,3 +181,22 @@ async def test_host_with_non_default_port_in_url():
             "authorization": "Basic dXNlcm5hbWU6cGFzc3dvcmQ=",
         }
     }
+
+
+@pytest.mark.asyncio
+async def test_none_headers():
```

Comment:
Should we add a small docstring to explain what this test is about? I think embedding the "Requests compatibility" motivation is important here.

```suggestion
async def test_none_headers():
    """
    For compatibility with Requests, `None` header values are allowed but ignored.
    """
```